pkg/platform/runtime/alternative.go

```diff
@@ -216,6 +217,18 @@ func (ai *AlternativeInstall) PreInstall() error {
 				return locale.WrapError(err, "err_rm_artf", "", "Could not remove old package file at {{.V0}}.", file)
 			}
 		}
+
+		for _, dir := range v.Dirs {
+			if !fileutils.DirExists(dir) {
+				continue
+			}
+			if fileutils.IsEmptyDirs(dir) {
```

Comment:
I still think it would be good to add a check for whether that `dir` is also in any of the artifacts that we have to keep.  But as that is such an edge case, it would probably also be okay to put this in a follow-up story.

Reply:
This is easy enough to put in this PR